src/dromajo_cosim.cpp

```diff
@@ -296,6 +306,46 @@ int dromajo_cosim_step(dromajo_cosim_state_t *state, int hartid, uint64_t dut_pc
         r->common.pending_exception = -1;
     }
 
+#ifdef GOLDMEM_INORDER
+    bool do_ld  = (dut_insn & 0x7F) == 0x03 || (dut_insn & 0x7F) == 0x07;
```

Comment:
Should at least have an assert to capture the fact that this doesn't support compressed instructions.

Also, we really should factor out all this decoding.  I'm pretty sure we do it many times.